api/src/main/java/org/openmrs/module/reporting/common/SqlRunner.java

```diff
@@ -144,6 +144,50 @@ public SqlResult executeSql(String sql, Map<String, Object> parameterValues) {
         return result;
 	}
 
+    /**
+     * Executes a Sql Script
+     */
+    public SqlIterator executeSqlToIterator(String sql) {
+
+        SqlIterator iterator = null;
+        log.info("Executing SQL...");
+
+        List<String> sqlStatements = new ArrayList<String>();
+        sqlStatements.addAll(parseSqlIntoStatements(sql));
+
+        Boolean originalAutoCommit = null;
+
+        try {
+            originalAutoCommit = connection.getAutoCommit();
+            connection.setAutoCommit(false);
+
+            for (String sqlStatement : sqlStatements) {
+                Statement statement = null;
+                try {
+                    statement = connection.createStatement();
+                    statement.setFetchSize(10);
+                    log.debug("Executing: {}", sqlStatement);
+                    statement.executeQuery(sqlStatement);
+                    ResultSet resultSet = statement.getResultSet();
+
+                    if (resultSet != null) {
+                        ResultSetMetaData rsmd = resultSet.getMetaData();
+                        iterator = new SqlIterator(rsmd, resultSet, connection, statement);
+                    }
+                } catch (Exception e) {
+                    closeStatement(statement);
```

Comment:
Just confirming - are you intentionally only closing the statement upon exception?  But intend to not close the statement otherwise?  So normal successful execution will leave the statement open?